templates/deployment.yaml

```diff
@@ -221,6 +237,59 @@ spec:
       {{- end }}
       {{- end }}
 
+      {{- if .Values.vxlanController.enabled }}
+      - name: vxlan-controller-agent
+        image: {{ .Values.vxlanController.image.repository }}:{{ .Values.vxlanController.image.tag }}
+        imagePullPolicy: {{ .Values.vxlanController.image.pullPolicy | quote }}
+        securityContext:
+          capabilities:
+            add:
+            - NET_ADMIN
+
+      {{- if or .Values.vxlanController.ip .Values.vxlanController.staticRoutes }}
+      - name: {{ .Chart.Name }}-vxlan-controller-ip-setup
+        image: {{ .Values.networkUtility.image.repository }}:{{ .Values.networkUtility.image.tag }}
+        imagePullPolicy: {{ .Values.networkUtility.image.pullPolicy | quote }}
+        securityContext:
+          capabilities:
+            add: ["NET_ADMIN"]
+        command:
+        - sh
+        - -c
+        - |
+          set -ex
+            sleep 5
+          {{- range $element := .Values.vxlanController.ip }}
+            {{- if eq $element.type "bridge" }}
+            brctl addbr {{ $element.interface }}
+            {{- range $bind := $element.bind }}
+            brctl addif {{ $element.interface }} {{ $bind }}
+            ip link set {{ $element.interface }} up
+            {{- end }}
+            {{- end }}
+          {{- end }}
+          {{- range $element := .Values.vxlanController.ip }}
+            {{- if eq $element.type "ip" }}
+            ip addr add {{ $element.addr }} dev {{ $element.interface }}
+            {{- end }}
+            {{- if eq $element.type "interface" }}
+              {{- if eq $element.action "up" }}
+              ip link set {{ $element.interface }} up
+              {{- else if eq $element.action "down" }}
+              ip link set {{ $element.interface }} down
+              {{- end }}
+            {{- end }}
+          {{- end }}
+          {{- range $element := .Values.vxlanController.staticRoutes }}
+            ip route add {{ $element }}
+
+          {{- end }}
+
+          while true; do
```

Comment:
This should be an init-container, right? Is there any reason for making it a container instead?